xapian-core/backends/remote/remote-database.cc

```diff
@@ -906,7 +907,7 @@ RemoteDatabase::open_synonym_termlist(const string& word) const
     string message;
     send_message(MSG_SYNONYMTERMLIST, word);
     get_message(message, REPLY_SYNONYMTERMLIST);
-    return new RemoteMetadataTermList(word, std::move(message));
+    return new VectorTermList(std::move(message));
```

Comment:
Oh, looking at the individual commits I see you tried `RemoteMetadataTermList` first.

Testing that version, I see it fails because `RemoteMetadataTermList::get_approx_size()` isn't implemented.  You can just return `1` there - that's what `GlassMetadataTermList::get_approx_size()` does:

```
Xapian::termcount
GlassMetadataTermList::get_approx_size() const
{
    // Very approximate!  This is only used to build a balanced or tree, so
    // at least we'll get an even tree by returning a constant answer.
    return 1;
}
```

Also I think you want `string()` instead of `word` when creating the `RemoteMetadataTermList` here - the first argument is any common prefix, but in general there isn't a common prefix here.